Missing mathconstants.hpp - can't install

Hey Davis,

Wanted to mess around with this but having issues installing with R 3.4.4 on a 64-bit linux system. Looks like it's missing a `source/ql/mathconstants.hpp`?

``` r
> devtools::install_github("DavisVaughan/calendarrr")
Downloading GitHub repo DavisVaughan/calendarrr@master
from URL https://api.github.com/repos/DavisVaughan/calendarrr/zipball/master
Installing calendarrr
'/usr/lib/R/bin/R' --no-site-file --no-environ --no-save --no-restore --quiet CMD INSTALL  \
  '/tmp/RtmpvrbyJD/devtools531f15c54bbc/DavisVaughan-calendarrr-dd7bfba' --library='/home/claytonjy/R/x86_64-pc-linux-gnu-library/3.4'  \
  --install-tests 

* installing *source* package ‘calendarrr’ ...
** libs
g++  -I/usr/share/R/include -DNDEBUG -I . -I ../inst/include -I"/home/claytonjy/R/x86_64-pc-linux-gnu-library/3.4/Rcpp/include" -I"/home/claytonjy/R/x86_64-pc-linux-gnu-library/3.4/BH/include"    -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c cal_identifiers.cpp -o cal_identifiers.o
In file included from ./ql/errors.hpp:28:0,
                 from ./ql/time/calendar.hpp:29,
                 from cal_identifiers.cpp:1:
./ql/qldefines.hpp:126:32: fatal error: ql/mathconstants.hpp: No such file or directory
compilation terminated.
/usr/lib/R/etc/Makeconf:168: recipe for target 'cal_identifiers.o' failed
make: *** [cal_identifiers.o] Error 1
ERROR: compilation failed for package ‘calendarrr’
* removing ‘/home/claytonjy/R/x86_64-pc-linux-gnu-library/3.4/calendarrr’
Installation failed: Command failed (1)
```

Created on 2018-08-23 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).
(not really; messages didn't show up in reprex so I did that interactively)
<details><summary>Session info</summary>

``` r
devtools::session_info()
#> Session info -------------------------------------------------------------
#>  setting  value                       
#>  version  R version 3.4.4 (2018-03-15)
#>  system   x86_64, linux-gnu           
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_US.UTF-8                 
#>  tz       America/Detroit             
#>  date     2018-08-23
#> Packages -----------------------------------------------------------------
#>  package   * version date       source        
#>  backports   1.1.2   2017-12-13 CRAN (R 3.4.4)
#>  base      * 3.4.4   2018-03-16 local         
#>  compiler    3.4.4   2018-03-16 local         
#>  curl        3.2     2018-03-28 CRAN (R 3.4.4)
#>  datasets  * 3.4.4   2018-03-16 local         
#>  devtools    1.13.6  2018-06-27 CRAN (R 3.4.4)
#>  digest      0.6.15  2018-01-28 CRAN (R 3.4.4)
#>  evaluate    0.11    2018-07-17 CRAN (R 3.4.4)
#>  git2r       0.23.0  2018-07-17 CRAN (R 3.4.4)
#>  graphics  * 3.4.4   2018-03-16 local         
#>  grDevices * 3.4.4   2018-03-16 local         
#>  htmltools   0.3.6   2017-04-28 CRAN (R 3.4.4)
#>  httr        1.3.1   2017-08-20 CRAN (R 3.4.4)
#>  knitr       1.20    2018-02-20 CRAN (R 3.4.4)
#>  magrittr    1.5     2014-11-22 CRAN (R 3.4.4)
#>  memoise     1.1.0   2017-04-21 CRAN (R 3.4.4)
#>  methods   * 3.4.4   2018-03-16 local         
#>  R6          2.2.2   2017-06-17 CRAN (R 3.4.4)
#>  Rcpp        0.12.18 2018-07-23 CRAN (R 3.4.4)
#>  rmarkdown   1.10    2018-06-11 CRAN (R 3.4.4)
#>  rprojroot   1.3-2   2018-01-03 CRAN (R 3.4.4)
#>  stats     * 3.4.4   2018-03-16 local         
#>  stringi     1.2.4   2018-07-20 CRAN (R 3.4.4)
#>  stringr     1.3.1   2018-05-10 CRAN (R 3.4.4)
#>  tools       3.4.4   2018-03-16 local         
#>  utils     * 3.4.4   2018-03-16 local         
#>  withr       2.1.2   2018-03-15 CRAN (R 3.4.4)
#>  yaml        2.2.0   2018-07-25 CRAN (R 3.4.4)
```

</details>

--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -11,8 +11,9 @@ License: GPL (>= 2)
 Imports:
     glue,
     Rcpp,
-    R6
+    R6,
+    methods
 LinkingTo: Rcpp, BH
-RoxygenNote: 6.0.1
+RoxygenNote: 6.1.0
 URL: https://github.com/DavisVaughan/calendarrr
 BugReports: https://github.com/DavisVaughan/calendarrr/issues

--- a/src/ql/mathconstants.hpp
+++ b/src/ql/mathconstants.hpp
@@ -1,0 +1,130 @@
+/* -*- mode: c++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*
+ Copyright (C) 2010 StatPro Italia srl
+
+ This file is part of QuantLib, a free-software/open-source library
+ for financial quantitative analysts and developers - http://quantlib.org/
+
+ QuantLib is free software: you can redistribute it and/or modify it
+ under the terms of the QuantLib license.  You should have received a
+ copy of the license along with this program; if not, please email
+ <quantlib-dev@lists.sf.net>. The license is also available online at
+ <http://quantlib.org/license.shtml>.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the license for more details.
+*/
+
+#ifndef quantlib_math_constants_hpp
+#define quantlib_math_constants_hpp
+
+#include <cmath>
+
+#ifndef M_E
+    #define M_E         2.71828182845904523536
+#endif
+
+#ifndef M_LOG2E
+    #define M_LOG2E     1.44269504088896340736
+#endif
+
+#ifndef M_LOG10E
+    #define M_LOG10E    0.434294481903251827651
+#endif
+
+#ifndef M_IVLN10
+    #define M_IVLN10    0.434294481903251827651
+#endif
+
+#ifndef M_LN2
+    #define M_LN2       0.693147180559945309417
+#endif
+
+#ifndef M_LOG2_E
+    #define M_LOG2_E    0.693147180559945309417
+#endif
+
+#ifndef M_LN10
+    #define M_LN10      2.30258509299404568402
+#endif
+
+#ifndef M_PI
+#    define M_PI 3.141592653589793238462643383280
+#endif
+
+#ifndef M_TWOPI
+    #define M_TWOPI     (M_PI * 2.0)
+#endif
+
+#ifndef M_PI_2
+    #define M_PI_2      1.57079632679489661923
+#endif
+
+#ifndef M_PI_4
+    #define M_PI_4      0.785398163397448309616
+#endif
+
+#ifndef M_3PI_4
+    #define M_3PI_4     2.3561944901923448370E0
+#endif
+
+#ifndef M_SQRTPI
+    #define M_SQRTPI    1.77245385090551602792981
+#endif
+
+#ifndef M_1_PI
+    #define M_1_PI      0.318309886183790671538
+#endif
+
+#ifndef M_2_PI
+    #define M_2_PI      0.636619772367581343076
+#endif
+
+#ifndef M_1_SQRTPI
+    #define M_1_SQRTPI  0.564189583547756286948
+#endif
+
+#ifndef M_2_SQRTPI
+    #define M_2_SQRTPI  1.12837916709551257390
+#endif
+
+#ifndef M_SQRT2
+    #define M_SQRT2     1.41421356237309504880
+#endif
+
+#ifndef M_SQRT_2
+    #define M_SQRT_2    0.7071067811865475244008443621048490392848359376887
+#endif
+
+#ifndef M_SQRT1_2
+    #define M_SQRT1_2   0.7071067811865475244008443621048490392848359376887
+#endif
+
+#ifndef M_LN2LO
+    #define M_LN2LO     1.9082149292705877000E-10
+#endif
+
+#ifndef M_LN2HI
+    #define M_LN2HI     6.9314718036912381649E-1
+#endif
+
+#ifndef M_SQRT3
+    #define M_SQRT3     1.73205080756887719000
+#endif
+
+#ifndef M_INVLN2
+    #define M_INVLN2    1.4426950408889633870E0
+#endif
+
+/* This should ensure that no macro are redefined if we happen to
+   include <math.h> again, whether or not we're using our macros
+   or theirs. We can't know in advance, since it depends on the
+   order of inclusion of headers in client code. */
+#ifdef _MSC_VER
+    #undef _USE_MATH_DEFINES
+#endif
+
+#endif
+

--- a/src/ql/time/frequency.hpp
+++ b/src/ql/time/frequency.hpp
@@ -1,0 +1,58 @@
+/* -*- mode: c++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*
+ Copyright (C) 2004, 2005, 2006 Ferdinando Ametrano
+ Copyright (C) 2006 Katiuscia Manzoni
+ Copyright (C) 2000, 2001, 2002, 2003 RiskMap srl
+ Copyright (C) 2003, 2004, 2005, 2006, 2008 StatPro Italia srl
+
+ This file is part of QuantLib, a free-software/open-source library
+ for financial quantitative analysts and developers - http://quantlib.org/
+
+ QuantLib is free software: you can redistribute it and/or modify it
+ under the terms of the QuantLib license.  You should have received a
+ copy of the license along with this program; if not, please email
+ <quantlib-dev@lists.sf.net>. The license is also available online at
+ <http://quantlib.org/license.shtml>.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the license for more details.
+*/
+
+/*! \file frequency.hpp
+    \brief Frequency enumeration
+*/
+
+#ifndef quantlib_frequency_hpp
+#define quantlib_frequency_hpp
+
+#include <ql/qldefines.hpp>
+#include <iosfwd>
+
+namespace QuantLib {
+
+    //! Frequency of events
+    /*! \ingroup datetime */
+    enum Frequency { NoFrequency = -1,     //!< null frequency
+                     Once = 0,             //!< only once, e.g., a zero-coupon
+                     Annual = 1,           //!< once a year
+                     Semiannual = 2,       //!< twice a year
+                     EveryFourthMonth = 3, //!< every fourth month
+                     Quarterly = 4,        //!< every third month
+                     Bimonthly = 6,        //!< every second month
+                     Monthly = 12,         //!< once a month
+                     EveryFourthWeek = 13, //!< every fourth week
+                     Biweekly = 26,        //!< every second week
+                     Weekly = 52,          //!< once a week
+                     Daily = 365,          //!< once a day
+                     OtherFrequency = 999  //!< some other unknown frequency
+    };
+
+    /*! \relates Frequency */
+    std::ostream& operator<<(std::ostream& out,
+                             Frequency f);
+
+}
+
+#endif

